sympy/physics/control/tests/test_lti.py

```diff
@@ -577,53 +574,186 @@ def test_Feedback_construction():
     assert tf4 / (TransferFunction(1, 1, p) + tf4*tf6) == Feedback(tf4, tf6)
     assert tf5 / (tf + tf5) == Feedback(tf5, tf)
 
-    raises(ValueError, lambda: tf1*tf2*tf3 / (1 + tf1*tf2*tf3))
-    raises(ValueError, lambda: tf1*tf2*tf3 / tf3*tf5)
-    raises(ValueError, lambda: tf2*tf3 / (tf + tf2*tf3*tf4))
+    raises(ValueError, lambda: TF1*TF2*TF3 / (1 + TF1*TF2*TF3))
+    raises(ValueError, lambda: TF1*TF2*TF3 / TF3*tf5)
+    raises(ValueError, lambda: TF2*TF3 / (tf + TF2*TF3*tf4))
 
-    assert Feedback(tf, tf1*tf2*tf3).doit() == \
+    assert Feedback(tf, TF1*TF2*TF3).doit() == \
         TransferFunction((a2*s + p)*(s**2 + 2*s*wn*zeta + wn**2), k*(a2*p - s) + \
         (a2*s + p)*(s**2 + 2*s*wn*zeta + wn**2), s)
-    assert Feedback(tf1, tf2*tf3).doit() == \
+    assert Feedback(TF1, TF2*TF3).doit() == \
         TransferFunction((a2*s + p)*(s**2 + 2*s*wn*zeta + wn**2), (k*(a2*p - s) + \
         (a2*s + p)*(s**2 + 2*s*wn*zeta + wn**2))*(s**2 + 2*s*wn*zeta + wn**2), s)
-    assert Feedback(tf1*tf2, tf5).doit() == \
+    assert Feedback(TF1*TF2, tf5).doit() == \
         TransferFunction(k*(a0 + s)*(s**2 + 2*s*wn*zeta + wn**2), (k*(-a0 + a1*s**2 + a2*s) + \
         (a0 + s)*(s**2 + 2*s*wn*zeta + wn**2))*(s**2 + 2*s*wn*zeta + wn**2), s)
     assert Feedback(tf4, tf6).doit() == \
         TransferFunction(p*(p + s)*(a0*p + p**a1 - s), p*(p*(p + s) + (-p + s)*(a0*p + p**a1 - s)), p)
     assert -Feedback(tf4*tf6, TransferFunction(1, 1, p)).doit() == \
         TransferFunction(-p*(-p + s)*(p + s)*(a0*p + p**a1 - s), p*(p + s)*(p*(p + s) + (-p + s)*(a0*p + p**a1 - s)), p)
 
-    assert Feedback(tf1, tf2*tf5).rewrite(TransferFunction) == \
+    assert Feedback(TF1, TF2*tf5).rewrite(TransferFunction) == \
         TransferFunction((a0 + s)*(s**2 + 2*s*wn*zeta + wn**2), (k*(-a0 + a1*s**2 + a2*s) + \
         (a0 + s)*(s**2 + 2*s*wn*zeta + wn**2))*(s**2 + 2*s*wn*zeta + wn**2), s)
     assert Feedback(TransferFunction(1, 1, p), tf4).rewrite(TransferFunction) == \
         TransferFunction(p, a0*p + p + p**a1 - s, p)
+
+
+def test_TransferFunctionMatrix_construction():
+    tf5 = TransferFunction(a1*s**2 + a2*s - a0, s + a0, s)
+    tf4 = TransferFunction(a0*p + p**a1 - s, p, p)
+    tf6 = TransferFunction(a0*s**3 + a1*s**2 - a2*s, b0*p**4 + b1*p**3 - b2*s*p, s)
+    tf7 = TransferFunction(p, a0, p)
+
+    tfm1 = TransferFunctionMatrix([TF1, TF2], (2, 1), s)
+    assert tfm1.shape == (tfm1.outputs, tfm1.inputs) == (2, 1)
+    assert tfm1.args == ([TF1, TF2], (2, 1), s)
+    assert tfm1.var == s
+
+    tfm2 = TransferFunctionMatrix([-TF1, TF2], (2, 1), s)
+    assert tfm2.shape == (tfm2.outputs, tfm2.inputs) == (2, 1)
+    assert tfm2.args == ([-TF1, TF2], (2, 1), s)
+    assert tfm2.var == s
+
+    tfm3 = TransferFunctionMatrix([tf7], (1, 1), p)
+    assert tfm3.shape == (tfm3.outputs, tfm3.inputs) == (1, 1)
+    assert tfm3.var == p
+    assert tfm3.args == ([tf7], (1, 1), p)
+
+    tfm4 = TransferFunctionMatrix([TF3, tf5, tf6], (3, 1), s)
```

Comment:
Why does the user have to pass in the shape to initialize? Can't this be inferred from the dimensions of the list of transferfunctions? SymPy Matrix and NumPy array infer these, for example.

Reply:
Please see https://github.com/sympy/sympy/pull/19761#discussion_r463996029.

Reply:
I don't see how that comment answers my question. I don't see why the shape arg or the var arg are necessary when initializing a TransferFunctionMatrix. This information is already present (implicitly) in the first argument.

Reply:
Well, that's what I did in the first place. I guess I have to revert the commit because `shape` and `var` attributes can be taken out from the first argument.

Reply:
Why should you have to drop parameters if there are alternative ways like making them as optional parameters

Reply:
>  Well, that's what I did in the first place. I guess I have to revert the commit because shape and var attributes can be taken out from the first argument.

What was the reason for adding them?

Reply:
There are some reasons from matrix computation perspective that shape parameter cannot always be dropped because it cannot represent the shape of (0, 0), (0, n), (n, 0), ..., which are often useful for describing zero rank matrix.

Reply:
I'm not sure I understand that. sympy.Matrix doesn't require declaring the shape on instantiation, so why would this matrix-like object need to?

Reply:
Sympy matrix automatically looks up the shape from the lists if they are not explicitly stated, but stores the shape in the Basic.args
Same thing can be done for transfer function matrix.

Reply:
> Same thing can be done for transfer function matrix

That sounds good to me!